controllers/machinedeletionremediation_controller.go

```diff
@@ -625,3 +647,18 @@ func (r *MachineDeletionRemediationReconciler) getMachineOwnerSpecReplicas(ctx c
 	}
 	return int(replicas), nil
 }
+
+func (r *MachineDeletionRemediationReconciler) getMachineKindAndNamespace(ctx context.Context) (kind, namespace string, err error) {
+	machines := &machinev1beta1.MachineList{}
+	if err = r.List(ctx, machines); err != nil {
+		r.Log.Error(err, "could not list Machines")
+		return "", "", err
+	}
+
+	if len(machines.Items) == 0 {
+		err = fmt.Errorf("Could not get Machine Namespace: no Machines found")
```

Comment:
The error is a little bit confusing `Could not get Machine Namespace: no Machines found`.
The word `Namespace` is needed here?
Did you maybe mean  `Could not get Machine: no Machines found` or  `Could not get Machine in machine namespace: no Machines found` or `Could not find any machine in openshift-machine-api namespace`?

Reply:
`Namespace` is the target of the function, together with `Kind`.

`getMachineKindAndNamespace` function returns the `Kind` and `Namespace` of Machine type, in case the Namespace is not (not sure whether that can be possible) `openshift-machine-api` and `Kind` is not "Machine".

Reply:
> Namespace is not (not sure whether that can be possible) openshift-machine-api

+1 I think the machines live only at `openshift-machine-api` namespace

Reply:
What made me doubt here is why do we print that we can't get the machine's namespace when there are no machines ( `len(machines.Items) == 0`)? How it is related? 
A better check could be whether we can list/find the machine's namespace or `machines.Items[0].Namespace != nil` or if `openshift-machine-api` was created (what should be trivial IIUC in OCP).

Reply:
If there is no need to pick `Namespace` and `Kind` at runtime, I can remove this function.

> What made me doubt here is why do we print that we can't get the machine's namespace when there are no machines ( len(machines.Items) == 0)? How it is related?

It picks the Namespace from one of the Machines, so if it cannot find any Machine it cannot get the Namespace (I expect this happens only due to a temporary error, and it will retry in the next Reconcile loop). Maybe I'm not understanding your point

Reply:
I removed this logic as not needed